MediaBrowser.Common/Net/NetworkExtensions.cs

```diff
@@ -9,240 +15,341 @@ namespace MediaBrowser.Common.Net
         {
-            if (source.Count == 0 || excludeList is null)
+            if (host.IsEmpty)
             {
-                return new Collection<IPObject>(source);
+                addresses = Array.Empty<IPAddress>();
+                return false;
             }
 
-            Collection<IPObject> results = new Collection<IPObject>();
+            host = host.Trim();
 
-            bool found;
-            foreach (var outer in source)
+            // See if it's an IPv6 with port address e.g. [::1] or [::1]:120.
+            if (host[0] == '[')
             {
-                found = false;
+                int i = host.IndexOf("]", StringComparison.Ordinal);
+                if (i != -1)
+                {
+                    return TryParseHost(host[1..(i - 1)], out addresses);
+                }
+
+                addresses = Array.Empty<IPAddress>();
+                return false;
+            }
 
-                foreach (var inner in excludeList)
+            var hosts = new List<string>();
+            foreach (var splitSpan in host.Split(':'))
+            {
+                hosts.Add(splitSpan.ToString());
+            }
+
+            if (hosts.Count <= 2)
+            {
+                // Is hostname or hostname:port
+                if (_fqdnRegex.IsMatch(hosts[0]))
                 {
-                    if (outer.Equals(inner))
+                    try
                     {
-                        found = true;
-                        break;
+                        addresses = Dns.GetHostAddresses(hosts[0]);
+                        return true;
+                    }
+                    catch (SocketException)
+                    {
+                        // Log and then ignore socket errors, as the result value will just be an empty array.
+                        Console.WriteLine("GetHostAddresses failed.");
                     }
                 }
 
-                if (!found)
+                // Is an IP4 or IP4:port
+                host = hosts[0].Split('/')[0];
+
+                if (IPAddress.TryParse(host, out var address))
                 {
-                    results.AddItem(outer, isNetwork);
+                    if (((address.AddressFamily == AddressFamily.InterNetwork) && (!isIPv4Enabled && isIPv6Enabled)) ||
+                        ((address.AddressFamily == AddressFamily.InterNetworkV6) && (isIPv4Enabled && !isIPv6Enabled)))
+                    {
+                        addresses = Array.Empty<IPAddress>();
+                        return false;
+                    }
+
+                    addresses = new[] { address };
+
+                    // Host name is an ip4 address, so fake resolve.
+                    return true;
+                }
+            }
+            else if (hosts.Count <= 9) // 8 octets + port
+            {
+                var splitSpan = host.Split('/');
+                if (splitSpan.MoveNext() && IPAddress.TryParse(splitSpan.Current, out var address))
```

Comment:
It will only be false if the span is empty. The if on line 321 should check that hosts isn't empty  and then you can use LeftPart